### PR TITLE
Merge probe and app binaries, run with ./scope probe or ./scope app.

### LINKED ISSUE #569: Scope binaries are getting a bit big
```
-rwxr-xr-x 1 vagrant vagrant  22M Oct 17 12:36 scope-app
-rwxr-xr-x 1 vagrant vagrant  21M Oct 17 12:36 scope-probe
```

Making them one binary would probably save most of this, as I bet its mostly shared libraries.

### PR DESCRIPTION
**Pull request by tomwilkie**
The benefits here are (1) 50% reduction in build time and (2) almost 50% reduction in container size.

Fixes #569

### CODE REVIEW

**Comment by paulbellamy** [comment]
Maybe move `prog/(app|probe)/main.go` to `prog/(app|probe).go`, to remove some unnecessary directory nesting.

File: prog/main.go [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+import (
+	"fmt"
+	"os"
+
+	"github.com/weaveworks/scope/prog/app"
+	"github.com/weaveworks/scope/prog/probe"
+)
+
+func usage() {
+	fmt.Printf("usage: %s (app|probe) args...\n", os.Args[0])
```
Comment (paulbellamy) [inline prog/main.go]: usage normally goes to stderr

File: prog/app/main.go [hunk 2]
```diff
@@ -6,11 +6,13 @@ import (
 	"log"
 	"math/rand"
 	"net/http"
-	_ "net/http/pprof"
 	"strconv"
 	"strings"
 	"time"
 
+	// a blank import should be only in a main or test package, or have a comment justifying it
+	_ "net/http/pprof"
```
Comment (paulbellamy) [inline prog/app/main.go]: Seems unrelated?

**Comment by tomwilkie** [comment]
@paulbellamy PTAL
